src/lib.rs

```diff
@@ -316,8 +388,36 @@ macro_rules! impl_compare {
 }
 
 impl_compare!(&'b str, &'a str);
+impl_compare!(CompleteStr<'b>, &'a str);
 impl_compare!(&'b [u8], &'a [u8]);
+impl_compare!(CompleteByteSlice<'b>, &'a [u8]);
 impl_compare!(&'b [u8], &'a str);
+impl_compare!(CompleteByteSlice<'b>, &'a str);
+
+impl<A: Compare<B>, B> Compare<LocatedSpan<B>> for LocatedSpan<A> {
+    #[inline(always)]
+    fn compare(&self, t: LocatedSpan<B>) -> CompareResult {
+        self.fragment.compare(t.fragment)
+    }
+
+    #[inline(always)]
+    fn compare_no_case(&self, t: LocatedSpan<B>) -> CompareResult {
+        self.fragment.compare_no_case(t.fragment)
+    }
+}
```

Comment:
Probably worth to add unit tests so we're sure that it works as expected. What do you think?

(well, that's not really unit testing as we don't use mocks, but still :))